utils/s2n_array.c

```diff
@@ -64,6 +67,16 @@ struct s2n_array *s2n_array_new(uint32_t element_size)
     return array;
 }
 
+S2N_RESULT s2n_array_init(struct s2n_array *array, uint32_t element_size)
+{
+    ENSURE_REF(array);
+
+    CHECKED_MEMSET(array, 0, sizeof(struct s2n_array));
+    array->element_size = element_size;
+
+    return S2N_RESULT_OK;
+}
```

Comment:
Could you add a proof harness for this function as well? All others functions for `s2n_array` have one. It should be pretty similar to the one for `s2n_array_new`.

Reply:
Talked offline. I'm going to add the proof for s2n_array_init in a separate PR in a (probably already failed) attempt to keep this PR manageable. Issue: https://github.com/awslabs/s2n/issues/2415